python/ray/dag/compiled_dag_node.py

```diff
@@ -746,6 +743,15 @@ def _get_node_id(self):
             for idx in task.downstream_node_idxs:
                 frontier.append(idx)
 
+        from ray.dag.constants import RAY_ADAG_ENABLE_VERIFY_GRAPH
+
+        if RAY_ADAG_ENABLE_VERIFY_GRAPH:
+            assert self._verify_graph(), (
```

Comment:
`_verify_graph` relies on `self.actor_to_tasks`, so this is the earliest point for graph verification.